src/sdk/namespace_fs.js

```diff
@@ -2419,15 +2448,35 @@ class NamespaceFS {
             try {
                 upload_params = await this._start_upload(fs_context, undefined, undefined, params, 'w');
 
-                // the delete marker path built from its version info (mtime + ino)
+                let file_path;
                 const stat = await upload_params.target_file.stat(fs_context);
-                delete_marker_version_id = this._get_version_id_by_stat(stat);
-                const file_path = this._get_version_path(params.key, delete_marker_version_id);
+                if (this._is_versioning_enabled()) {
+                    // the delete marker path built from its version info (mtime + ino)
+                    delete_marker_version_id = this._get_version_id_by_stat(stat);
+                    file_path = this._get_version_path(params.key, delete_marker_version_id);
+                } else {
+                    // the delete marker file name would be with a 'null' suffix
+                    delete_marker_version_id = NULL_VERSION_ID;
+                    file_path = this._get_version_path(params.key, delete_marker_version_id);
+                }
```

Comment:
let's move ` file_path = this._get_version_path(params.key, delete_marker_version_id);` outside of the conditions